lib/paths/index.ts

```diff
@@ -94,6 +104,17 @@ type UnsafePaths<Type, Options extends Required<PathsOptions>> = Type extends Ty
     : never
   : never;
 
+type AnyArrayIndexAccessor<Type, Options extends Required<PathsOptions>> = Type extends any[]
+  ? Options["anyArrayIndexAccessor"] extends string
+    ? `${Options["anyArrayIndexAccessor"]}`
+    : false
+  : false;
+
+type AnyArrayIndexAccessorOrKey<
+  Key extends Pathable,
+  Options extends RecursivePathsOptions,
+> = Options["anyArrayIndexAccessor"] extends false ? Key : Key | Options["anyArrayIndexAccessor"];
+
```

Comment:
You can safely remove boolean conditional types since the config option is always a string.